File: trabalho arquitetura de software/csharp/projeto1/entidades/Hotel.cs
using System;
using System.Collections.Generic;

namespace Dominios {
	
	public class Hotel 
	{
		// atributos primitivos adicionais
		public string nome;
		public string endereco;
				
		//atributos pedidos no projeto
		// 1:1
		public string cadeiadohotel;
		public Conta contadeparceiro;
		
		//1:N
		public List<Oferta> ofertas;
		public List<Acomodacao> acomodacoes;
		
		public Hotel
		(
			string nome,
			string endereco,
			string cadeiadohotel,
			List<Oferta> ofertas,
			List<Acomodacao> acomodacoes
		)
		{
			this.nome = nome;
			this.endereco = endereco;
			this.cadeiadohotel = cadeiadohotel;
			this.ofertas = ofertas;
			this.acomodacoes = acomodacoes;
		}
		
	}
}

File: trabalho arquitetura de software/python/projeto/hotel.py
from dataclasses import dataclass
from typing import Optional
from datetime import datetime
from oferta import Oferta;
from acomodacao import Acomodacao;

 
@dataclass
class Hotel:
    nome: str
    endereco: str
    cadeiadohotel: Optional[str] = None 
    contadeparceiro: str
    ofertas: list[Oferta]
    acomodacoes: list[Acomodacao]